server/lib/statsd-collector.js

```diff
@@ -10,6 +10,72 @@ var uaParser = require('ua-parser');
 
 var STATSD_PREFIX = 'fxa.content.';
 
+function getGenericTags(body) {
+  /*jshint maxcomplexity:7 */
+  // see more about tags here: http://docs.datadoghq.com/guides/metrics/
+  var tags = [
+    'campaign:' + body.campaign,
+    'context:' + body.context,
+    'broker:' + body.broker,
+    'entrypoint:' + body.entrypoint,
+    'migration:' + body.migration,
+    'lang:' + body.lang,
+    'service:' + body.service
+  ];
+
+  if (body.screen) {
+    var screen = body.screen;
+
+    tags = tags.concat([
+      'screen_client_width:' + screen.clientWidth,
+      'screen_client_height:' + screen.clientHeight,
+      'screen_device_pixel_ratio:' + screen.devicePixelRatio,
+      'screen_width:' + screen.width,
+      'screen_height:' + screen.height
+    ]);
+  }
+
+  if (body.agent) {
+    var agent = uaParser.parse(body.agent);
+    if (agent) {
+      if (agent.ua) {
+        tags = tags.concat([
+          'agent_ua_name:' + agent.ua.toString(), // -> "Safari 5.0.1"
+          'agent_ua_family:' + agent.ua.family, // -> "Safari"
+          'agent_ua_version:' + agent.ua.toVersionString(), // -> "5.0.1"
+          'agent_ua_version_major:' + agent.ua.major // -> "5"
+        ]);
+      }
+
+      if (agent.os) {
+        tags = tags.concat([
+          'agent_os_name:' + agent.os.toString(), // -> "iOS 5.1"
+          'agent_os_version:' + agent.os.toVersionString(), // -> "5.1"
+          'agent_os_family:' + agent.os.family, // -> "iOS"
+          'agent_os_major:' + agent.os.major // -> "5"
+        ]);
+      }
+
+      if (agent.device) {
+        tags = tags.concat([
+          'agent_device:' + agent.device.family // -> "iPhone"
+        ]);
+      }
+    }
+  }
+
+  return tags;
+}
+
+
+function getImpressionTags(impression) {
+  return [
+    'marketing_campaignId:' + impression.campaignId,
```

Comment:
Ah, I did not read their blurb closely enough. Thanks for pointing that out.

An yep, `marketing_campaign_id` is better than `marketing_campaignid`. Updating.